src/primitive/util.rs

```diff
@@ -183,6 +204,34 @@ mod tests {
         assert_eq!(point, get_max_point(triangle.iter(), &direction, &t));
     }
 
+    #[test]
+    fn test_closest_point_2d() {
+        let start = Vector2::new(3., -1.);
+        let end = Vector2::new(-1., 3.);
+        let point = Vector2::zero();
+        let p = get_closest_point_on_edge(&start, &end, &point);
+        assert_ulps_eq!(1f32, p.x);
```

Comment:
should probably do it on points right away, e.g. `assert_ulps_eq!(p, Point2::new(1f32, 1f32))`